src/area.cpp

```diff
@@ -3516,9 +3513,8 @@ CMDF( do_aset )
          ch->print( "Usage: aset <filename> gem <percentage>\r\n" );
          return;
       }
-      value = atoi( argument.c_str(  ) );
-      tarea->tg_gem = value;
-      ch->printf( "Area chance to generate gem set to %hu%%\r\n", value );
+      tarea->tg_gem = stoi( argument );
+      ch->printf( "Area chance to generate gem set to %hu%%\r\n", tarea->tg_ge );
```

Comment:
I think the printf function is using the wrong variable here, should probably be tarea->tg_gem